core/actionProxy/actionproxy.py

```diff
@@ -127,18 +127,29 @@ def error(msg):
 
         try:
             input = json.dumps(args)
-            p = subprocess.Popen(
-                [self.binary, input],
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-                env=env)
+            if len(input) > 131071:             # MAX_ARG_STRLEN (131071) linux/binfmts.h
+                # pass argument via stdin
+                p = subprocess.Popen(
+                    [self.binary],
+                    stdin=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    env=env)
+            else:
+                # pass argument via stdin and command parameter
```

Comment:
I suggest that we _always_ do this. That is we migrate all actions to use the `stdin` path instead of `argv[1]`. The code above will be a fail safe for all existing actions which will only work up to 128K.

Reply:
I agree with this. `stdin` should be the default. Input can be passed as an argument as well but not if it's larger than 128K.